tests/functional/test_newsletter_embed.py

```diff
@@ -0,0 +1,61 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from pages.newsletter_embed import MozillaBasePage
+
+
+@pytest.mark.nondestructive
+@pytest.mark.parametrize('slug', [
+    pytest.mark.smoke(''),
+    ('about'),
+    pytest.mark.smoke('contribute'),
+    ('mission'),
+    ('firefox/all'),
+    ('firefox/sync'),
+    pytest.mark.smoke('firefox/desktop'),
+    ('firefox/desktop/customize'),
+    ('firefox/desktop/fast'),
+    ('firefox/desktop/trust'),
+    pytest.mark.smoke(('firefox/sync')),
+    pytest.mark.skip_if_not_firefox('plugincheck',
+        reason='https://bugzilla.mozilla.org/show_bug.cgi?id=1245208')
+    ('teach/smarton'),
+    pytest.mark.skip_if_not_firefox('teach/smarton/tracking',
+        reason='Newsletter is only shown to Firefox browsers.'),
+    pytest.mark.skip_if_not_firefox('teach/smarton/security',
+        reason='Newsletter is only shown to Firefox browsers.'),
+    pytest.mark.skip_if_not_firefox('teach/smarton/surveillance',
+        reason='Newsletter is only shown to Firefox browsers.')])
+def test_newsletter_default_values(slug, base_url, selenium):
+    page = MozillaBasePage(base_url, selenium, slug=slug).open()
```

Comment:
Rather than introduce `MozillaBasePage` I would make the parameters something like `page_object` and `url_kwargs`. Then you can pass `HomePage`, `AboutPage`, etc. and not have to duplicate the URL paths. The `url_kwargs` would be something like `{'slug': 'teach/smarton'}` and expanded when instantiating the page object.